packages/@divvi/mobile/src/earn/EarnWithdrawConfirmationScreen.tsx

```diff
@@ -40,29 +46,69 @@ const TAG = 'earn/EarnWithdrawConfirmationScreen'
 
 type Props = NativeStackScreenProps<StackParamList, Screens.EarnWithdrawConfirmationScreen>
 
+function getTotalWithdrawAmountsPerToken<
+  T extends {
```

Comment:
can we avoid the generic type? seems like rewardTokens is something that is constructed in the `useRewards` helper and the only extra field is balance, can we just make it explicit?